### ISSUE TITLE
Tests or linting should test email formatting

### ISSUE THREAD

**Issue by di** [pypa/warehouse#13216]
In #13215, it was determined that some emails were failing to send due to being mis-formatted. While #13214 will catch these in production, our tests and/or linting should catch these before merge.

**Comment by miketheman** [comment]
Can you share any details on how the infinite retry was observed? That might help determine the testing/linting strategy.

**Comment by di** [comment]
Yep, this: https://github.com/pypi/warehouse/blob/5046d0b747566dbbb9504e35c36359688b2d89a5/warehouse/email/__init__.py#L66-L74
As far as I'm aware, that will keep retrying indefinitely for most exceptions.

**Comment by miketheman** [comment]
I spent some time on this today, trying to load and render the `subject.txt` templates, but it turns out that's not as simple as it looked at first glance.
The jinja `Template.render()` function needs a valid context that includes the variables **for each template**, so it's not readily generalizable as far as I can tell.

We might want to pursue a different approach, like a static linter rule.
